module/product/src/Application/Controller/Api/Relations/VariableProductAvailableChildrenAction.php

```diff
@@ -0,0 +1,168 @@
+     * @param VariableProductAvailableChildrenGrid $grid
+     */
+    public function __construct(
+        ProductChildrenQueryInterface $query,
+        GridRenderer $gridRenderer,
+        VariableProductAvailableChildrenGrid $grid
+    ) {
+        $this->query = $query;
+        $this->gridRenderer = $gridRenderer;
+        $this->grid = $grid;
+    }
+
+    /**
+     * @IsGranted("PRODUCT_READ")
+     *
+     * @SWG\Tag(name="Product")
+     * @SWG\Parameter(
+     *     name="product",
+     *     in="path",
+     *     type="string",
+     *     description="Product ID",
+     * )
+     * @SWG\Parameter(
+     *     name="limit",
+     *     in="query",
+     *     type="integer",
+     *     required=true,
+     *     default="50",
+     *     description="Number of returned lines",
+     * )
+     * @SWG\Parameter(
+     *     name="offset",
+     *     in="query",
+     *     type="integer",
+     *     required=true,
+     *     default="0",
+     *     description="Number of start line",
+     * )
+     * @SWG\Parameter(
+     *     name="field",
+     *     in="query",
+     *     required=false,
+     *     type="string",
+     *     enum={"sku","template", "default_label", "attach_flag"},
+     *     description="Order field",
+     * )
+     * @SWG\Parameter(
+     *     name="order",
+     *     in="query",
+     *     required=false,
+     *     type="string",
+     *     enum={"ASC","DESC"},
+     *     description="Order",
+     * )
+     * @SWG\Parameter(
+     *     name="columns",
+     *     in="query",
+     *     required=false,
+     *     type="string",
+     *     description="Columns"
+     * )
+     * @SWG\Parameter(
+     *     name="filter",
+     *     in="query",
+     *     required=false,
+     *     type="string",
+     *     description="Filter"
+     * )
+     * @SWG\Parameter(
+     *     name="view",
+     *     in="query",
+     *     required=false,
+     *     type="string",
+     *     enum={"grid","list"},
+     *     description="Specify respons format"
+     * )
+     * @SWG\Parameter(
+     *     name="language",
+     *     in="path",
+     *     type="string",
+     *     required=true,
+     *     default="en_GB",
+     *     description="Language Code",
+     * )
+     * @SWG\Response(
+     *     response=200,
+     *     description="Returns products",
+     * )
+     *
+     * @ParamConverter(class="Ergonode\Product\Domain\Entity\AbstractProduct")
+     * @ParamConverter(class="Ergonode\Grid\RequestGridConfiguration")
+     *
+     * @param AbstractProduct          $product
+     * @param Language                 $language
+     * @param RequestGridConfiguration $configuration
+     *
+     * @return Response
+     */
+    public function __invoke(
+        AbstractProduct $product,
+        Language $language,
+        RequestGridConfiguration $configuration
+    ): Response {
+        Assert::isInstanceOf($product, VariableProduct::class);
```

Comment:
There's just one issue with it. It seems like the exception thrown internally causes 500 server error. Maybe it's worth considering?